sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/CostBasedJoinReorder.scala

```diff
@@ -206,14 +206,14 @@ object JoinReorderDP extends PredicateHelper with Logging {
     // For the lower level k, we only need to search from 0 to lev - k, because when building
     // a join from A and B, both A J B and B J A are handled.
     while (k <= lev - k) {
-      val oneSideCandidates = existingLevels(k).values.toSeq
+      val oneSideCandidates = existingLevels(k).values.toSeq.sortBy(_.itemIds.head)
```

Comment:
Ah, is this the only way to make it deterministic? I was hoping we could make the test assertions deterministic with a sort without changing the underlying logic. This may have a performance penalty.

Reply:
need to try, there is no good idea now.